### ISSUE TITLE
Container builds take 4x as long following base image switch to Python 3.11

### ISSUE THREAD

**Issue by manics** [jupyterhub/binderhub#1636]
### Bug description

https://github.com/jupyterhub/binderhub/actions/workflows/publish.yml?query=branch%3Amain

Container images based on Python 3.9 used to take around 30 minutes total for chartpress to publish the chart.

Following the [switch to Python 3.11](https://github.com/jupyterhub/binderhub/actions/runs/3877664894) this job takes around 2 hours.

**Comment by consideRatio** [comment]
I bet it relates to compiling things that have pre-built wheels for py39 but not for py311

**Comment by consideRatio** [comment]
The single build step that takes notable time is [this one](https://github.com/jupyterhub/binderhub/actions/runs/4121415327/jobs/7117080666#step:10:1742), but not form amd64 but for arm64.

```
#29 [linux/arm64 build-stage 6/6] RUN --mount=type=cache,target=/tmp/pip-cache     pip install build  && pip wheel         .         pycurl         -r helm-chart/images/binderhub/requirements.txt
```

The slowness stems from this dependency:

https://github.com/jupyterhub/binderhub/blob/c71a2034aed6deb86935dbe9ce76304ad4f871e3/helm-chart/images/binderhub/requirements.txt#L59-L64

**Comment by minrk** [comment]
Maybe best to switch the image to 3.10 for now?

**Comment by consideRatio** [comment]
It seems that this is to be fixed quite soon, I suggest we wait. See https://github.com/grpc/grpc/issues/32454 and note that their current release candidate has aarch64 wheels for py311

**Comment by consideRatio** [comment]
- Closed by #1654 that bumps to a grpc version that has an aarch64 build for py311
